moto/elasticbeanstalk/models.py

```diff
@@ -148,5 +159,19 @@ def _find_environment_by_arn(self, arn: str) -> FakeEnvironment:
                     return env
         raise KeyError()
 
+    def delete_application(
+        self, application_name: str, terminate_env_by_force: bool
+    ) -> FakeApplication:
+        if application_name:
+            if application_name in self.applications:
+                application = self.applications[application_name]
+                application.status = "deleting"
```

Comment:
What's the reasoning behind setting the status, rather then just removing the application altogether?

The `describe_applications` call doesn't seem to return a status, so a user would not be able to tell the difference between an active application, and an application that was 'deleted' in the background.

How does this work in AWS - is the application still returned in `describe_applications` for another x minutes, before AWS actually deletes it?

Reply:
Thanks for the feedback! You're absolutely right, the `status` field shouldn't be included, I thought it was always included in all the resources with different naming conventions, but for the Elastic Beanstalk nothing is returned. I removed this field from the model.